app/controllers/concerns/course/forum/topic_controller_subscription_concern.rb

```diff
@@ -0,0 +1,37 @@
+module Course::Forum::TopicControllerSubscriptionConcern
+  extend ActiveSupport::Concern
+
+  def subscribe
+    if set_subscription_state
```

Comment:
I just think permissions of 'subscribe' should be the same as 'show'. If it not correct, then I will not do this. Also, I intend to add the abilities in a separate PR.